app/views/tools/_petition.html.erb

```diff
@@ -38,8 +37,7 @@
           <%= render 'tools/thank_you' -%>
         </div>
         <div class='unsigned'>
-          <%= form_for @signature, url: '/tools/petition',
-            html: {class: 'petition-tool'} do |f| -%>
+          <%= form_for @signature, url: tools_petition_path, html: {class: 'petition-tool'} do |f| -%>
```

Comment:
Just so I know, where is the tools_petition_path get set?

Reply:
all of these path helpers are created in [this file](https://github.com/EFForg/action-center-platform/blob/master/config/routes.rb#L12) (the line I linked to makes that one!)

`config/routes.rb` does more than that though -- it controls what URLs go to what controller methods. I often reference [this guide](https://guides.rubyonrails.org/routing.html) when I'm making routing changes.

Reply:
Cool, thank you!